awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -0,0 +1,552 @@
+    json_output = {'changed': False}
+    config_name = 'tower_cli.cfg'
+
+    def __init__(self, argument_spec, **kwargs):
+        args = dict(
+            tower_host=dict(required=False, fallback=(env_fallback, ['TOWER_HOST'])),
+            tower_username=dict(required=False, fallback=(env_fallback, ['TOWER_USERNAME'])),
+            tower_password=dict(no_log=True, required=False, fallback=(env_fallback, ['TOWER_PASSWORD'])),
+            validate_certs=dict(type='bool', aliases=['tower_verify_ssl'], required=False, fallback=(env_fallback, ['TOWER_VERIFY_SSL'])),
+            tower_oauthtoken=dict(type='str', no_log=True, required=False, fallback=(env_fallback, ['TOWER_OAUTH_TOKEN'])),
+            tower_config_file=dict(type='path', required=False, default=None),
+        )
+        args.update(argument_spec)
+        kwargs['supports_check_mode'] = True
+
+        super(TowerModule, self).__init__(argument_spec=args, **kwargs)
+
+        self.load_config_files()
+
+        # Parameters specified on command line will override settings in any config
+        if self.params.get('tower_host'):
+            self.host = self.params.get('tower_host')
+        if self.params.get('tower_username'):
+            self.username = self.params.get('tower_username')
+        if self.params.get('tower_password'):
+            self.password = self.params.get('tower_password')
+        if self.params.get('validate_certs') is not None:
+            self.verify_ssl = self.params.get('validate_certs')
+        if self.params.get('tower_oauthtoken'):
+            self.oauth_token = self.params.get('tower_oauthtoken')
+
+        # Perform some basic validation
+        if not re.match('^https{0,1}://', self.host):
+            self.host = "https://{0}".format(self.host)
+
+        # Try to parse the hostname as a url
+        try:
+            self.url = urlparse(self.host)
+        except Exception as e:
+            self.fail_json(msg="Unable to parse tower_host as a URL ({1}): {0}".format(self.host, e))
+
+        # Try to resolve the hostname
+        hostname = self.url.netloc.split(':')[0]
+        try:
+            gethostbyname(hostname)
+        except Exception as e:
+            self.fail_json(msg="Unable to resolve tower_host ({1}): {0}".format(hostname, e))
+
+        self.session = Request(cookies=self.cookie_jar)
+
+    def load_config_files(self):
+        # Load configs like TowerCLI would have from least import to most
+        config_files = [join('/etc/tower/', self.config_name), join(expanduser("~"), ".{0}".format(self.config_name))]
+        local_dir = getcwd()
+        config_files.append(join(local_dir, self.config_name))
+        while split(local_dir)[1]:
+            local_dir = split(local_dir)[0]
+            config_files.insert(2, join(local_dir, self.config_name))
+
+        for config_file in config_files:
+            try:
+                self.load_config(config_file)
+            except ConfigFileException:
```

Comment:
`tower-cli` throws an error if one of the default files exists but is not formatted properly, I think that behavior is better as it helps with debugging bad configs.